Source/AudioVisualizer.cpp

```
@@ -176,6 +205,66 @@ void AudioVisualizer::setAudioData(const float* leftChannel, const float* rightC
+    leftPeak_.store(leftPeak);
+    rightPeak_.store(rightPeak);
+    leftRMS_.store(leftRMS);
+    rightRMS_.store(rightRMS);
+
+    leftPeakHold_.store(std::max(leftPeakHold_.load(), leftPeak));
+    rightPeakHold_.store(std::max(rightPeakHold_.load(), rightPeak));
+
+    setDirty(true);
+}
+
+void AudioVisualizer::setInterleavedWaveform(const float* interleavedStereo, size_t numFrames) {
```

Clang (14.0.6)

[warning] parameter 'numFrames' is unused

(misc-unused-parameters)